app/src/main/java/edu/drexel/lapcounter/lapcounter/backend/Database/Workout/WorkoutRepository.java

```diff
@@ -55,8 +72,35 @@ public Workouts call() throws Exception {
 
     }
 
+    public ArrayList<Workout> getWorkoutsByDateRange(final Date start_time,final Date end_time)
+    {
+        ExecutorService ex = Executors.newSingleThreadExecutor();
+        Future<List<Workout>> res = ex.submit(new Callable<List<Workout>>() {
+            @Override
+            public List<Workout> call() throws Exception {
 
-    private static class insertAsyncTask extends AsyncTask<Workouts,Void,Void>
+                return mWorkoutDao.getWorkoutsByDateRange(start_time,end_time);
+            }
+        });
+        List<Workout> workouts = new ArrayList<Workout>();
+        try
+        {
+            workouts = res.get();
+        }
+        catch(Exception e)
+        {
+            Log.i("ERROR",e.toString());
```

Comment:
I'm not liking the look of this very general exception handler that demotes the error to an info level. If this fails in an unexpected way, it should fail loudly. I would recommend changing this method signature to `getWorkoutsByDateRange(...) throws WhateverDBExceptionRoomThrows` and let the caller handle the exception.